src/3rd_party/CMakeLists.txt

```diff
@@ -72,15 +76,19 @@ if(USE_SENTENCEPIECE)
 
   # regardless of -DUSE_STATIC_LIBS setting always build sentencepiece statically
   set(SPM_ENABLE_SHARED OFF CACHE BOOL  "Builds shared libaries in addition to static libraries." FORCE)
-  set(SPM_ENABLE_TCMALLOC ON CACHE BOOL "Enable TCMalloc if available.")
-
-  if(USE_STATIC_LIBS)
-    set(SPM_TCMALLOC_STATIC ON CACHE BOOL "Link static library of TCMALLOC." FORCE)
-  else(USE_STATIC_LIBS)
-    set(SPM_TCMALLOC_STATIC OFF CACHE BOOL "Link static library of TCMALLOC.")
-  endif(USE_STATIC_LIBS)
 
-  add_subdirectory(./sentencepiece)
+  if(USE_TCMALLOC)
+    set(SPM_ENABLE_TCMALLOC ON CACHE BOOL "Enable TCMalloc if available.")
+    if(USE_STATIC_LIBS)
+      set(SPM_TCMALLOC_STATIC ON CACHE BOOL "Link static library of TCMALLOC." FORCE)
+    else(USE_STATIC_LIBS)
+      set(SPM_TCMALLOC_STATIC OFF CACHE BOOL "Link static library of TCMALLOC.")
+    endif(USE_STATIC_LIBS)
+  else(USE_TCMALLOC)
+    set(SPM_ENABLE_TCMALLOC OFF CACHE BOOL "Enable TCMalloc if available.")
+  endif(USE_TCMALLOC)
+
+  add_subdirectory(./sentencepiece EXCLUDE_FROM_ALL)
```

Comment:
`EXCLUDE_FROM_ALL` is causing `spm_encode` to be not built (as marian only requires `libsentencepiece.a`).

@erip

Reply:
Great, so the tests can be updated to not rely on the binaries being built. I can fix that and hopefully this can get merged.